reconciler/retries.go

```diff
@@ -98,17 +112,27 @@ func (rq *retries) Add(obj any) {
 	}
 	item.object = obj
 	item.numRetries += 1
-	item.retryAt = time.Now().Add(rq.backoff.Duration(item.numRetries))
+	duration := rq.backoff.Duration(item.numRetries)
+	item.retryAt = time.Now().Add(duration)
 	heap.Push(&rq.queue, item)
+
+	// New item is at the top of the queue, reset the timer.
```

Comment:
nit: I don't think it's guaranteed that a new item is at the top after `Push`, but it's probably more efficient to reschedule anyway than trying to figure that out.